src/Custom/Responses/CreateResponseOptions.cs

```diff
@@ -0,0 +1,162 @@
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenAI.Responses
+{
+    [CodeGenSuppress("CreateResponseOptions", typeof(IEnumerable<ResponseItem>))]
+
+    [CodeGenType("DotNetCreateResponse")]
+    public partial class CreateResponseOptions
+    {
+         public CreateResponseOptions(string model, IEnumerable<ResponseItem> inputItems)
+        {
+            Argument.AssertNotNull(inputItems, nameof(inputItems));
+            Argument.AssertNotNullOrEmpty(model, nameof(model));
+
+            Metadata = new ChangeTrackingDictionary<string, string>();
+            Tools = new ChangeTrackingList<ResponseTool>();
+            InputItems = inputItems.ToList();
+            IncludedProperties = new ChangeTrackingList<IncludedResponseProperty>();
+            Model = model;
+        }
+
+        /// <summary>
+        /// Gets or sets whether to run the response in background mode. This corresponds to the "background" property in the JSON representation.
+        /// </summary>
+        [CodeGenMember("Background")]
+        public bool? BackgroundModeEnabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets how tool calls should be selected during response generation. This corresponds to the "tool_choice" property in the JSON representation.
+        /// </summary>
+        [CodeGenMember("ToolChoice")]
+        public ResponseToolChoice ToolChoice { get; set; }
+
+        /// <summary>
+        /// Gets or sets the input items to be processed for the response. This corresponds to the "input" property in the JSON representation.
+        /// </summary>
+        [CodeGenMember("Input")]
+        public IList<ResponseItem> InputItems { get; internal set; }
```

Comment:
Since this is a protocol model and this property is public, I don't think this needs an internal setter anymore.